### PR TITLE
feat: Replace all toasts with sonner

### PR DESCRIPTION
**Pull request by ivasilov**

### CODE REVIEW

**Comment by MildTomato** [comment]
sorry, will look at this later, been swamped

**Review by alaister — COMMENTED** [review]
small nit: we've lost the progress bar for file uploads – is it easy to re-add?
Screenshot 2024-08-02 at 13 00 08
Screenshot 2024-08-02 at 12 58 43

**Comment by joshenlim** [comment]
> small nit: we've lost the progress bar for file uploads – is it easy to re-add? Screenshot 2024-08-02 at 13 00 08 Screenshot 2024-08-02 at 12 58 43

also possible to make the toast slightly longer so that the description at the bottom doesnt wrap?

**Comment by ivasilov** [comment]
> small nit: we've lost the progress bar for file uploads – is it easy to re-add?

Fixed, it was using a wrong import.

> also possible to make the toast slightly longer so that the description at the bottom doesnt wrap?

To fix this, I had to make all toasts a bit wider. All toasts need to be the same width so that they stack nicely, otherwise they'll look like this:
Screenshot 2024-08-06 at 13 49 02

Screenshot 2024-08-06 at 13 49 09

With the bigger width, there's a lot of empty space, but they still look ok.

Screenshot 2024-08-06 at 13 55 59

**Comment by joshenlim** [comment]
just checking on the preview - the "Please do not close the browser until completed" text has quite low contrast imo
image
image

will bump it from text-foreground-muted to text-foreground-lighter

**Comment by joshenlim** [comment]
Otherwise LGTM! Tested locally
- Success, loading, error toasts from the dashboard

**Review by joshenlim — APPROVED** [review]

**Comment by joshenlim** [comment]
Just to check - i'm guessing this PR can be closed in favor of this then?
https://github.com/supabase/supabase/pull/28167

**Comment by joshenlim** [comment]
sorry 🤦 hit close with comment by accident

**Comment by MildTomato** [comment]
Screenshot 2024-08-08 at 1 51 05 PM

@ivandotv why don't we just align to the right? 

the super wide toasts don't look right to me.

**Comment by MildTomato** [comment]
I think we should also consider a different overlay entirely for file uploads, treat it more like Google Drive, so you can stack uploads on top of each other in the same overlay.
Screenshot 2024-08-08 at 1 56 41 PM

**Comment by MildTomato** [comment]
Screenshot 2024-08-08 at 1 58 46 PM

@ivasilov can we just say "Do not close browser", add the Cancel button in, and then the size doesn't need to be changed at all

File: packages/ui/src/components/shadcn/ui/toast-progress.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * like so
+ *  toast.loading(
+ *    <ToastProgress
+ *      progress={0}
+ *      message={`Downloading files...`}
+ *    />,
+ *    { id: toastId, closeButton: false }
+ *  )
+ */
+export const ToastProgress = ({
```
Comment (MildTomato) [inline packages/ui/src/components/shadcn/ui/toast-progress.tsx:15]: SonnerProgress

File: packages/ui/src/components/shadcn/ui/sonner.tsx [hunk 2]
```diff
@@ -19,7 +20,11 @@ const SonnerToaster = ({ ...props }: ToasterProps) => {
         info: <StatusIcon variant="default" />,
       }}
       theme={theme as ToasterProps['theme']}
-      className="toaster group"
+      // pointer-events-auto is needed to fix the toast when above radix modals. Set the width to 420px to fix the toast
+      // progress component (bottom row rendered in two lines).
+      className="toaster group pointer-events-auto w-[420px]"
```
Comment (MildTomato) [inline packages/ui/src/components/shadcn/ui/sonner.tsx:25]: can we remove this hardcoded w ?

File: packages/ui/src/components/shadcn/ui/sonner.tsx [hunk 3]
```diff
@@ -48,8 +53,21 @@ const SonnerToaster = ({ ...props }: ToasterProps) => {
             'group toast group-[.toaster]:!bg-warning-200 group-[.toaster]:!border-warning-500',
           error:
             'group toast group-[.toaster]:!bg-destructive-200 group-[.toaster]:!border-destructive-500',
+          closeButton: cn(
+            // unset all styles set from sonner
+            'absolute right-2 top-2 rounded-md text-foreground/50 opacity-0 transition-opacity',
+            'hover:text-foreground focus:opacity-100 focus:outline-none focus:ring-2 group-hover:opacity-100',
+            'group-[.destructive]:text-red-300 group-[.destructive]:hover:text-red-50 group-[.destructive]:focus:ring-red-400',
+            'group-[.destructive]:focus:ring-offset-red-600',
```
Comment (MildTomato) [inline packages/ui/src/components/shadcn/ui/sonner.tsx:61]: red ? you mean destructive ?

File: packages/ui/src/components/shadcn/ui/sonner.tsx [hunk 4]
```diff
@@ -48,8 +53,21 @@ const SonnerToaster = ({ ...props }: ToasterProps) => {
...
+            'left-auto transform-none bg-transparent border-0 border-transparent hover:!bg-transparent hover:border-transparent'
+          ),
+          content: 'grow',
+          //group-[.toaster]:bg-overlay group-[.toaster]:text-foreground group-[.toaster]:border group-[.toaster]:border-overlay
         },
+        closeButton: true,
```
Comment (MildTomato) [inline packages/ui/src/components/shadcn/ui/sonner.tsx:67]: why is there a close button by default ?

**Review by MildTomato — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/shadcn/ui/sonner.tsx (hunk 4 above)
Comment (ivasilov) [reply packages/ui/src/components/shadcn/ui/sonner.tsx:67]: Toasts in studio on production already have close buttons. I'm just matching the current implementation. Don't have a strong opinion on this.

**Comment by ivasilov** [comment]
> I think we should also consider a different overlay entirely for file uploads, treat it more like Google Drive, so you can stack uploads on top of each other in the same overlay. Screenshot 2024-08-08 at 1 56 41 PM

I've moved all progress toasts to bottom right. They look weird when you do multiple uploads in parallel and one of the uploads has completed. Parallel uploads is probably not that used and the user can still access/dismiss them by hovering over the stack.
Screenshot 2024-08-08 at 11 33 43

File: packages/ui/src/components/shadcn/ui/toast-progress.tsx (hunk 1 above)
Comment (ivasilov) [reply packages/ui/src/components/shadcn/ui/toast-progress.tsx]: Fixed.

File: packages/ui/src/components/shadcn/ui/sonner.tsx (hunk 3 above)
Comment (ivasilov) [reply packages/ui/src/components/shadcn/ui/sonner.tsx]: Fixed.

File: packages/ui/src/components/shadcn/ui/sonner.tsx (hunk 2 above)
Comment (ivasilov) [reply packages/ui/src/components/shadcn/ui/sonner.tsx]: Fixed.

**Review by MildTomato — APPROVED** [review]

**Review by MildTomato — CHANGES_REQUESTED** [review]
might be issue with how we're importing sonner into apps. checking...

**Review by MildTomato — APPROVED** [review]
lgtm

**Review by thorwebdev — APPROVED** [review]

**Pull request merged**